src/test/groovy/com/bmuschko/gradle/docker/tasks/image/DockerfileTest.groovy

```diff
@@ -2,52 +2,97 @@ package com.bmuschko.gradle.docker.tasks.image
 
 import spock.lang.Specification
 
+import java.util.logging.Level
+import java.util.logging.Logger
+
 import static com.bmuschko.gradle.docker.tasks.image.Dockerfile.*
 
 class DockerfileTest extends Specification {
+    private static Logger LOG = Logger.getLogger(DockerfileTest.class.getCanonicalName());
     def "Instruction String representation is built correctly"() {
+
         expect:
-        instructionInstance.keyword == keyword
-        instructionInstance.build() == builtInstruction
+        LOG.fine "testing " + [
+            instruction: instructionInstance.class,
+            becomes: [
+                keyword: expectedKeyword,
+                withBuildText: expectedBuiltInstruction,
+            ]
+        ]
+
+        def actual
+        try {
+            actual = instructionInstance.build()
+        }
+        catch(Exception e) {
+            LOG.info "exception catched with message `${e.message}`"
+            if(!expectedBuiltInstruction.equals(e.class)) {
+                LOG.log(Level.WARNING, e.stackTrace?.join('\n\t'))
+            }
+            actual = e.class
+        }
+
+        instructionInstance.keyword == expectedKeyword
+        actual == expectedBuiltInstruction
```

Comment:
What exactly is going on with this try/catch block?

I would also change the log to `Exception caught with message`